yt/yt/server/node/exec_node/slot_manager.cpp

```diff
@@ -247,6 +243,21 @@ TFuture<void> TSlotManager::InitializeEnvironment()
         Bootstrap_->GetConfig()->JobResourceManager->ResourceLimits->Cpu,
         GetIdleCpuFraction());
 
+    auto slotInitTimeout = DynamicConfig_.Acquire()->SlotInitTimeout;
+    for (int slotIndex = 0; slotIndex < SlotCount_; ++slotIndex) {
+        auto slotInitFuture = JobEnvironment_->InitSlot(slotIndex)
+            .WithTimeout(slotInitTimeout);
+        slotInitFuture.Subscribe(BIND([this, this_ = MakeStrong(this), slotIndex] (const TError& error) {
+            if (error.IsOK()) {
+                auto guard = WriterGuard(FreeSlotsSpinLock_);
+                FreeSlots_.push(slotIndex);
+            } else {
+                auto wrappedError = TError("Failed to initialize slot %v", slotIndex) << error;
+                JobEnvironment_->Disable(wrappedError);
```

Comment:
std::move()

Reply:
Resurrection has been working here in case of Porto job environment.
Now we may lose slot forever. Let's add retries here